components/Search/SearchForm.jsx

```diff
@@ -0,0 +1,60 @@
+import React, { Component } from 'react';
+
+import TextField from 'material-ui/TextField';
+import RaisedButton from 'material-ui/RaisedButton';
+import ActionSearch from 'material-ui/svg-icons/action/search';
+import CircularProgress from 'material-ui/CircularProgress';
+
+import style from './search.css';
+
+const loadingStyle = {
+  display: 'block',
+  margin: '40px auto 0 auto',
+};
+
+class SearchForm extends Component {
+
+  render() {
+    const { hintText, onSubmit, disabled } = this.props;
+    const formSubmit = (e) => {
+      e.preventDefault();
+      onSubmit(this.textField.input.value);
+    };
+    const loadingIndicator = () => (
+      <div className={style.loadingWrapper}>
+        <CircularProgress style={loadingStyle} />
+      </div>
+    );
+
+    return (
+      <form onSubmit={formSubmit}>
+        <TextField
+          ref={(ref) => {
+            this.textField = ref;
+          }}
+          disabled={disabled}
+          hintText={hintText}
+          fullWidth
+        />
+        <RaisedButton
+          type="submit"
+          label="Search"
+          disabled={disabled}
+          icon={<ActionSearch />}
+        />
+        {disabled === true ? loadingIndicator() : ''}
+      </form>
+    );
+  }
+}
+
+SearchForm.propTypes = {
+  onSubmit: React.PropTypes.func,
+  hintText: React.PropTypes.string,
+};
+SearchForm.defaultProps = {
+  onSubmit: () => {},
+  hintText: '',
+};
+
```

Comment:
Do we need default values? I assume the Function will throw an error if it's undefined on submit but undefined string should be OK.